docs/enhancements/add-external-providers.md

```diff
@@ -0,0 +1,187 @@
+[isVirtual](https://github.com/openshift/assisted-installer-agent/blob/master/src/inventory/system_vendor.go#L15-L19]
+list and that it can be detected using the fields in 
+[SystemVendor](https://github.com/openshift/assisted-service/blob/master/models/system_vendor.go).
+
+#### Changes to the Openshift installer project:
+
+A small change is required in the installer project.
+Since the openshift installer is generating a random InfraID the names of the Nodes(in the manifest) are different
+from the nodes the user provisions, and since we have a provider set then the provider will try to start new machines 
+due to the naming mismatch.
+Since the InfraID is random we can't tell the user to create the Nodes with specific names.
+Also, the InfraID is later used by the platforms in various ways to identify the cluster,
+for example in oVirt we use it to set an oVirt tag which helps us group the cluster resources in oVirt,
+and we rely on it in various places such as cluster destroy or CSI/Machine provider logic to filter cluster resources
+quickly and prevent unnecessary API calls.
+
+We need to add an env var to allow us to override the generated InfraID.
+We started a PR on the [installer project](https://github.com/openshift/installer/pull/5101).
+
+### Risks and Mitigations
+
+As with every large code change, this change contains the risk of breaking existing functionality.
+This can be mitigated by sanity testing and making sure each provider implements unit tests and e2e test suite.
+
+## Design Details
+
+### Open Questions
+
+1. Throught the assisted installer code we have structs and functions that are specific to baremetal(at least from the name),
+- Are they safe to be extended to support providers?
+- Is there anything baremetal specific we should be conserned about?
+- Is the assisted installer team aims to modify the names to be generic? for example I don't see why the
+[InstallerConfig](https://github.com/openshift/assisted-service/blob/master/internal/installcfg/installcfg.go#L75-L110)
+ struct is BareMetal.
+ 
+ Some examples but it is on every file:
+- [BMACReconciler](https://github.com/openshift/assisted-service/blob/master/internal/controller/controllers/bmh_agent_controller.go#L50-L55)
+- [bareMetalInventory](https://github.com/openshift/assisted-service/blob/master/internal/bminventory/inventory.go#L149-L175)
+- [InstallerConfigBaremetal](https://github.com/openshift/assisted-service/blob/master/internal/installcfg/installcfg.go#L75-L110)
+- [bmhIsMaster](https://github.com/openshift/assisted-service/blob/master/internal/ignition/ignition.go#L541-L551)
+
+2. Is there a use case in which the user will provision the Nodes on a certain provider but want to disable
+provider integration? meaning nodes will be provider will be discovered but we want platform none anyway?
+in which case what would be the best way to handle this in terms of user interaction?
+would an env var flag to disable provider integration is enough?
```

Comment:
This seems like a per-cluster setting, not something we'd enable or disable for the entire application.

Reply:
Correct.  Currently, with the vSphere integration, we detect the hosts' platforms and if they are all vSphere, we report via API that vSphere platform integration is possible.  However, the user must choose whether to enable it or not.

Reply:
Probably answers my question above :sweat_smile: 
Is there a vSphere integration demo around?

Reply:
@eliorerz Can you please provide a link to the demo that you recorded?